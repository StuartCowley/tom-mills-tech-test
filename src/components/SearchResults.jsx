import React from "react";
import "../styles/search-results.css";
import PropTypes from "prop-types";

const SearchResults = ({ results }) => {
  if(!results.length) {
    return <p className="no-results">No results</p>
  } else {
    return (
      <div className="search-results" >
        {results.map((imageData) => (
          <img
            className="card-image"
            data-testid="test-result"
            src={imageData.links[0].href}
            alt={imageData.data[0].title}
            key={imageData.data[0].nasa_id}
          />
        ))}
      </div>
    );
  };
};

export default SearchResults;

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape({})),
}