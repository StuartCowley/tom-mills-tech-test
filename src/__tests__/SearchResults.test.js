import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";
import testData from "./testdata.json";

describe("SearchResults", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults results={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders No Results if empty array is passed as a prop", () => {
    render(<SearchResults results={[]} />);

    expect(screen.getByText("No results")).toBeInTheDocument();
  });

  it("renders test data", () => {
    const { getAllByTestId } = render(<SearchResults results={testData} />);

    expect(getAllByTestId("test-result")).toHaveLength(3);
  });
});
