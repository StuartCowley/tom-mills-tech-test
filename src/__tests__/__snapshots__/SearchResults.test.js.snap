// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <p
    class="no-results"
  >
    No results
  </p>
</DocumentFragment>
`;
